tensorboard/webapp/hparams/_redux/hparams_reducers.ts

```diff
@@ -87,7 +89,82 @@ const reducer: ActionReducer<HparamsState, Action> = createReducer(
         metrics: nextMetricFilters,
       },
     };
-  })
+  }),
+  on(actions.dashboardHparamColumnAdded, (state, {column, nextTo, side}) => {
+    const {dashboardDisplayedHparamColumns: oldColumns} = state;
+
+    let destinationIndex = oldColumns.length; // Default to append at end.
+    if (nextTo !== undefined && side !== undefined) {
+      const nextToIndex = oldColumns.findIndex(
+        (col) => col.name === nextTo.name
+      );
+      if (nextToIndex !== -1) {
+        destinationIndex = side === Side.RIGHT ? nextToIndex + 1 : nextToIndex;
+      }
+    }
+    let newColumn = {...column, enabled: true};
+    let newColumns = [...oldColumns];
+    newColumns.splice(destinationIndex, 0, newColumn);
+
+    return {
+      ...state,
+      dashboardDisplayedHparamColumns: newColumns,
+    };
+  }),
+  on(actions.dashboardHparamColumnRemoved, (state, {column}) => {
+    const newColumns = state.dashboardDisplayedHparamColumns.filter(
+      ({name}) => name !== column.name
+    );
+
+    return {
+      ...state,
+      dashboardDisplayedHparamColumns: newColumns,
+    };
+  }),
+  on(actions.dashboardHparamColumnToggled, (state, {column: toggledColumn}) => {
+    const newColumns = state.dashboardDisplayedHparamColumns.map((column) => {
+      if (column.name === toggledColumn.name) {
+        return {
+          ...column,
+          enabled: !toggledColumn.enabled,
+        };
+      } else {
+        return column;
+      }
+    });
+
+    return {
+      ...state,
+      dashboardDisplayedHparamColumns: newColumns,
+    };
+  }),
+  on(
+    actions.dashboardHparamColumnOrderChanged,
+    (state, {source, destination, side}) => {
+      const {dashboardDisplayedHparamColumns: columns} = state;
+      let sourceIndex = columns.findIndex(
+        (column: ColumnHeader) => column.name === source.name
+      );
+      let destinationIndex = columns.findIndex(
+        (column: ColumnHeader) => column.name === destination.name
+      );
+      if (sourceIndex === -1 || sourceIndex === destinationIndex) {
+        return state;
+      }
+      if (destinationIndex === -1) {
+        destinationIndex = side === Side.LEFT ? 0 : columns.length - 1;
+      }
+
+      const newColumns = [...columns];
+      newColumns.splice(sourceIndex, 1);
+      newColumns.splice(destinationIndex, 0, source);
```

Comment:
If the destination IS found, the `side` is not taken into account.

Heads up, if `side` IS accounted for you will likely run into an issue with way you derive the indexes, then mutate the array.

Reply:
> If the destination IS found, the side is not taken into account.

Yeah this is WAI: in typical cases, we just swap source with dest - no Side required. Side is essentially used as a backup for when dest is not found. This allows us to do things like: ![reorder](https://github.com/tensorflow/tensorboard/assets/736199/928e5d52-2140-41cc-9960-13301b70ec36)

I'll add additional conditional logic to make sure this only triggers when dest is not found AND side exists, and will add a comment to clarify this behavior.

PLMK if this seems unnecessarily convoluted - I'm also open to simply removing the side param and simply doing nothing if dest isn't found. (IMO reordering elements to the end of the "column group" should be rather intuitive though)